Ermumau work06

EMumau opened the pull request:
Fix for issue #44.  This also depends on PR #427 in OMI.
The Script Provider uses the same shared library for every provider.  There are some global variables that were being reassigned at run time when multiple providers where run concurrently.  This would cause errors is provider A was queried, then provider B was queried, and then provider A again before provider A had unloaded.  I changed the signature of the start fn that OMI calls to use a member of OMI to store this instead of using a global variable in the provider.
This also changes from using AF_UNIX sockets to AF_INET sockets.

Commit:
Stage some testing changes.

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master'

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master'

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master'

JumpingYang001 commented on provider/server.cpp:
```diff
@@ -233,24 +239,370 @@ handle_return (
+                    result = Server::SUCCESS;
+                }
+#if (PRINT_BOOKENDS)
+                else
```
need to set result for other 3 else condition?

EMumau replied:
Result is initialized to a fail condition.  The case where it is set is the single success case.  This is as designed.

JumpingYang001 commented on provider/server.cpp:
```diff
@@ -233,24 +239,370 @@ handle_return (
+            {
+                // failed to set the socket to non-blocking
+                SCX_BOOKEND_PRINT ("failed to set new socket to non-blocking");
+                close (fd);
```
break here?

EMumau replied:
This is a case that ~should not~ ever occur.  And if it does, it is not a terminal condition.  There could be an incoming connection that fails.  This handles the failed connection and continues to receive other incoming connections.  This is as designed.

JumpingYang001 reviewed (commented):
add 2 comments, thanks.

Commit:
This should fix issue #44.  This changes is also dependent on PR #427 in OMI.
The Script Provider uses the same shared library for every provider.  There are some global variables that were being reassigned at run time when multiple providers where run concurrently.  This would cause errors is provider A was queried, then provider B was queried, and then provider A again before provider A had unloaded.  I changed the signature of the start fn that OMI calls to use a member of OMI to store this instead of using a global variable in the provider.
This also changes from using AF_UNIX sockets to AF_INET sockets.

JumpingYang001 commented:
ok, BTW, looks like Travis failed.

EMumau commented:
The omi branch for build-omi_script-provider needs to be updated to include changes through sha 8adb210 -> https://github.com/Microsoft/omi/commit/8adb210a06cdb84bd3c99f5f65298d69b2c18081

JumpingYang001 commented:
That's what I get from Travis log:
```
 instance of Full_Test

2513> {

2514>     [Key] Id=3

2515>     BooleanTest=false

2516>     RealTest=10

2517>     IntTest=255

2518>     StringTest=Max value for ASCII Symbols

2519Enumerate failed!

2520../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2521../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

2522instance of OMI_Error

2523{

2524    OwningEntity=OMI:CIMOM

2525    MessageID=OMI:MI_Result:5

2526    Message=The specified class does not exist.

2527    MessageArguments={}

2528    PerceivedSeverity=7

2529    ProbableCause=0

2530    ProbableCauseDescription=Unknown

2531    CIMStatusCode=5

2532    OMI_Code=5

2533    OMI_Category=10

2534    OMI_Type=MI

2535    OMI_ErrorMessage=The specified class does not exist.

2536}

2537../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2538../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

25391c1

2540< instance of OMI_Error

2541---

2542> instance of Full_Test

25433,14c3,20

2544<     OwningEntity=OMI:CIMOM

2545<     MessageID=OMI:MI_Result:5

2546<     Message=The specified class does not exist.

2547<     MessageArguments={}

2548<     PerceivedSeverity=7

2549<     ProbableCause=0

2550<     ProbableCauseDescription=Unknown

2551<     CIMStatusCode=5

2552<     OMI_Code=5

2553<     OMI_Category=10

2554<     OMI_Type=MI

2555<     OMI_ErrorMessage=The specified class does not exist.

2556---

2557>     [Key] Id=0

2558>     BooleanTest=true

2559>     RealTest=3.1415

2560>     StringTest=Value of PI

2561> }

2562> instance of Full_Test

2563> {

2564>     [Key] Id=1

2565>     BooleanTest=false

2566>     IntTest=365

2567>     StringTest=Days in a year

2568> }

2569> instance of Full_Test

2570> {

2571>     [Key] Id=2

2572>     BooleanTest=false

2573>     RealTest=1.618

2574>     StringTest=Golden ratio

2575Delete instance failed!

2576../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2577../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

2578instance of OMI_Error

2579{

2580    OwningEntity=OMI:CIMOM

2581    MessageID=OMI:MI_Result:5

2582    Message=The specified class does not exist.

2583    MessageArguments={}

2584    PerceivedSeverity=7

2585    ProbableCause=0

2586    ProbableCauseDescription=Unknown

2587    CIMStatusCode=5

2588    OMI_Code=5

2589    OMI_Category=10

2590    OMI_Type=MI

2591    OMI_ErrorMessage=The specified class does not exist.

2592}

2593../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2594../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

2595instance of OMI_Error

2596{

2597    OwningEntity=OMI:CIMOM

2598    MessageID=OMI:MI_Result:5

2599    Message=The specified class does not exist.

2600    MessageArguments={}

2601    PerceivedSeverity=7

2602    ProbableCause=0

2603    ProbableCauseDescription=Unknown

2604    CIMStatusCode=5

2605    OMI_Code=5

2606    OMI_Category=10

2607    OMI_Type=MI

2608    OMI_ErrorMessage=The specified class does not exist.

2609}

2610../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2611../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

2612instance of OMI_Error

2613{

2614    OwningEntity=OMI:CIMOM

2615    MessageID=OMI:MI_Result:5

2616    Message=The specified class does not exist.

2617    MessageArguments={}

2618    PerceivedSeverity=7

2619    ProbableCause=0

2620    ProbableCauseDescription=Unknown

2621    CIMStatusCode=5

2622    OMI_Code=5

2623    OMI_Category=10

2624    OMI_Type=MI

2625    OMI_ErrorMessage=The specified class does not exist.

2626}

2627../../../omi/Unix/output/bin/omicli: result: MI_RESULT_INVALID_CLASS

2628../../../omi/Unix/output/bin/omicli: result: The specified class does not exist.

26291,15d0

2630< instance of OMI_Error

2631< {

2632<     OwningEntity=OMI:CIMOM

2633<     MessageID=OMI:MI_Result:5

2634<     Message=The specified class does not exist.

2635<     MessageArguments={}

2636<     PerceivedSeverity=7

2637<     ProbableCause=0

2638<     ProbableCauseDescription=Unknown

2639<     CIMStatusCode=5

2640<     OMI_Code=5

2641<     OMI_Category=10

2642<     OMI_Type=MI

2643<     OMI_ErrorMessage=The specified class does not exist.

2644< }

2645Delete all instances failed!

2646rm: cannot remove ‘*.pyc’: No such file or directory

2647Full CLI Test failed!

2648!!!!!!!!!!!!!!!!!!!!!!!

2649!!!!!!!!!!!!!!!!!!!!!!!

2650!!!CLI tests failed!!!!

2651!!!!!!!!!!!!!!!!!!!!!!!

2652!!!!!!!!!!!!!!!!!!!!!!!

2653>>> "./test_all" failed with 1 <<<

2654************************

2655************************

2656************************

2657***                  ***

2658***  Regress failed  ***

2659***                  ***

2660************************

2661************************

2662************************

2663

2664

2665The command "./regress" exited with 1.

2666

2667Done. Your build exited with 1.

```

alinbalutoiu commented:
Can you also squash the commits please? Thanks!
